ColumnNavigationMap Update

davidsklar99 opened the pull request:
ColumnNavigationMap 
- now allows input of array of column ids and display only those columns
- Arg added to Storybook

DetritalZirconCompilation
- uses new ColumnNavigationMap

davenquinn commented on packages/column-views/src/maps/column-navigation/mapbox/column-navigation-map.stories.ts:
```diff
@@ -25,6 +25,7 @@ export default {
 export const ColumnNavigation = {
```
Rather than updating this story, it'd be better to create a new one. That way you can ensure that the current behavior doesn't change when you add the new filtering capability.

davenquinn commented on packages/column-views/src/maps/column-navigation/mapbox/state.ts:
```diff
@@ -76,13 +77,15 @@ export function ColumnNavigationProvider({
-  const _columns = columns ?? useMacrostratColumns(projectID, inProcess);
+  const newColumns = getColsData({columns});
+  const _columns = columns ? newColumns : useMacrostratColumns(projectID, inProcess);
+
```
Instead of getting macrostrat columns separately, just filter the existing columns on the client side. Right now you are getting both, which is unnecessary.

davenquinn commented on packages/column-views/src/maps/column-navigation/mapbox/state.ts:
```diff
@@ -107,3 +110,16 @@ export function useColumnNavigationStore(
+
+function getColsData({columns}) {
+  let res = useAPIResult(
+    "https://macrostrat.org/api/v2/columns?col_id=" + columns?.join(",") + "&response=long&format=geojson"
```
This function should be deleted but also note that it's not usually ideal to hard-code a Macrostrat URL in this kind of location, as we might want to run this in different environments

davenquinn commented on packages/column-views/stories/facets/detrital-zircon.stories.ts:
```diff
@@ -109,3 +124,22 @@ export function DetritalZirconCompilation(defaultArgs) {
+    format: "geojson",
+    response: "light",
+  })
+  return useAPIResult("https://macrostrat.org/api/v2/measurements", {
```
Not sure how this is being nested in.

davenquinn commented on packages/column-views/stories/facets/detrital-zircon.stories.ts:
```diff
@@ -109,3 +124,22 @@ export function DetritalZirconCompilation(defaultArgs) {
   );
 }
+
+function getLayer(props) {
```
Seems like there might be a better way to arrange this that could better use react's capabilities

davenquinn reviewed (changes requested):
Overall, it's going to be necessary to adjust this to not break current functionality and work better with the data loader approach being used by the `useMacrostratData` hook

davenquinn commented on packages/column-views/src/maps/column-navigation/mapbox/state.ts:
```diff
@@ -76,13 +76,19 @@ export function ColumnNavigationProvider({
   //   store.setState({ columns: _columns, selectedColumn });
   // }, [projectID, inProcess, columns, getColumns]);
 
-  const _columns = columns ?? useMacrostratColumns(projectID, inProcess);
```
We should have both the `columns` and `columnIDs` options available.

- `columns` allows you to pass in a fully separate set of columns as GeoJSON (this is useful for instance for columns that have been pre-filtered or haven't yet been saved to the server)
- `columnIDs` would allow you to pass in a set of IDs to filter based on.
Alternatively, you could allow `columns` to be either an array of numbers or GeoJSON and change things based on which is passed.

davenquinn commented on packages/column-views/src/maps/column-navigation/mapbox/state.ts:
```diff
@@ -26,7 +26,7 @@ export interface NavigationProviderProps {
   selectedColumn?: number | null;
   hoveredColumn?: number | null;
-  columns?: ColumnGeoJSONRecordWithID[] | null;
+  columns?: number[] | null;
```
Don't change the behavior of an existing method! This could be done but would be a major breaking change.

Instead, either add a `number[]` option to the existing type or create a new `columnIDs` or `filteredColumnIDs` option

davenquinn commented on packages/column-views/src/maps/column-navigation/mapbox/state.ts:
```diff
@@ -76,13 +76,19 @@ export function ColumnNavigationProvider({
+  // filter columns if specified
+  if (columns?.length > 0) {
+    _columns = _columns?.filter((d) => columns.includes(d.id));
+  }
```
Consider a `useMemo` here so the filter doesn't need to run on every render.

davenquinn commented on packages/column-views/stories/facets/detrital-zircon.stories.ts:
```diff
@@ -67,39 +71,42 @@ export const BighornBasinColored = {
 };
 
 export function DetritalZirconCompilation(defaultArgs) {
+  const [mapInstance, setMapInstance] = useState(null);
```
the correct way to do this would be to use the `useMapRef` function

davenquinn commented on packages/column-views/stories/facets/detrital-zircon.stories.ts:
```diff
@@ -67,39 +71,42 @@ export const BighornBasinColored = {
+            selectedColumn: columnArgs?.col_id,
+            margin: 0,
+            accessToken: import.meta.env.VITE_MAPBOX_API_TOKEN,
+            onMapLoaded: handleMapLoaded, 
```
I'm kind of surprised we need all of this. Perhaps I can look over it?

davenquinn reviewed (changes requested):
Overall, this is getting there but there are a few things that need to be changed. Should I go ahead and update the PR or let you try a few more things?

davidsklar99 commented:
I fixed the ColumnNavigationMap so now it can take columns ids or geojson as input. I cleaned up the code for the DeltaZirconCompilation but still need you to take a look at it to see why the layer doesn't get added
